src/components/Message/MessageText.js

```diff
@@ -30,27 +30,38 @@ const MessageTextComponent = (props) => {
     unsafeHTML,
     customOptionProps,
   } = props;
+
   const reactionSelectorRef = useRef(
     /** @type {HTMLDivElement | null} */ (null),
   );
+
   const { onMentionsClick, onMentionsHover } = useMentionsUIHandler(message, {
     onMentionsClick: propOnMentionsClick,
     onMentionsHover: propOnMentionsHover,
   });
+
   const {
     onReactionListClick,
     showDetailedReactions,
     isReactionEnabled,
   } = useReactionClick(message, reactionSelectorRef);
+
+  const { client } = useContext(ChatContext);
   const { t } = useContext(TranslationContext);
+
   const hasReactions = messageHasReactions(message);
   const hasAttachment = messageHasAttachments(message);
   const handleReaction = useReactionHandler(message);
-  const messageTextItem = message?.text;
+  const connectedUserLanguage = client.user?.language;
+
+  const messageTextToRender =
+    message?.i18n?.[`${connectedUserLanguage}_text`] || message?.text;
```

Comment:
let's toss a test in here that renders a message with the i18n object and make sure it renders the translation and not message.text